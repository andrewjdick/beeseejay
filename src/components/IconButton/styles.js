import styled from "styled-components";

export const Button = styled.button`
  background-color: transparent;
  border: none;
  outline: none;
  padding: 0;
`;
