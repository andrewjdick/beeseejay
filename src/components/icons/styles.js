import styled from "styled-components";

export const Svg = styled.svg`
  color: rgba(0, 0, 0, 0.6);
  transition: all 200ms ease-in;

  &:hover {
    color: rgba(0, 0, 0, 0.8);
  }

  &:active {
    transform: scale(0.9);
    color: rgb(0, 0, 0);
  }
`;
