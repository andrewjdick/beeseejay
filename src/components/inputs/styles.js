import styled, { css } from "styled-components";
import { media } from "media.js";

const baseInputStyles = css`
  outline: none;
  border: none;
  padding: 5px;
  background-color: transparent;
  transition: box-shadow 100ms linear;

  &:focus {
    border: 1px solid grey;
    border-radius: 2px;
    box-shadow: 0px 0px 5px 2px rgba(0, 0, 0, 0.1);
  }

  &:hover {
    box-shadow: 0px 0px 5px 2px rgba(0, 0, 0, 0.1);
  }
`;

export const StyledInput = styled.input`
  ${baseInputStyles};

  text-overflow: ellipsis;
  font-weight: bold;
  font-size: 1.5rem;
  line-height: 1.6rem;

  ${media.tablet`
    font-size: 1rem;
    line-height: 1.1rem;
  `}
`;

export const StyledTextArea = styled.textarea`
  ${baseInputStyles};

  font-size: 1.1rem;
  line-height: 1.2rem;

  ${media.tablet`
    font-size: 0.8rem;
    line-height: 0.9rem;
  `}

  ${media.desktop`
    font-size: 0.75rem;
    line-height: 0.85rem;
  `}
`;
