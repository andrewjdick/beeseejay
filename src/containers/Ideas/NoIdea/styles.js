import styled from "styled-components";

export const Container = styled.div`
  flex: 1;
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const InnerContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;

export const Title = styled.h2`
  margin-bottom: 1em;
  font-weight: bold;
  font-size: 1.3rem;
  line-height: 1.5rem;
`;

export const Subtitle = styled.h3``;
