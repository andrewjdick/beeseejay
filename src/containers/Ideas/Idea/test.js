import React from "react";
import renderer from "react-test-renderer";
import { render, fireEvent } from "react-testing-library";
import "react-testing-library/cleanup-after-each";
import "jest-dom/extend-expect";

import { Idea } from ".";
import { CHECKMARK_WIDTH } from "./styles";

const IDEA_PROPS = {
  id: "1234-abcd-5678-efgh-9101",
  dateCreated: 1549985336574,
  isUpdated: false,
  title: "Test title",
  description: "Test description"
};

// Snapshot testing
it("Idea card renders correctly", () => {
  const input = renderer.create(<Idea {...IDEA_PROPS} />).toJSON();
  expect(input).toMatchSnapshot();
});

// Integration testing
it("displays the update notification when isUpdated is set to true", () => {
  const { rerender, queryByTestId } = render(<Idea {...IDEA_PROPS} />);
  const updatedProps = {
    ...IDEA_PROPS,
    isUpdated: true
  };

  expect(queryByTestId("update-success")).toHaveAttribute("width", "0");

  rerender(<Idea {...updatedProps} />);

  expect(queryByTestId("update-success")).toHaveAttribute(
    "width",
    `${CHECKMARK_WIDTH}`
  );
});

it("renders the correct elements", () => {
  const { queryByTestId } = render(<Idea {...IDEA_PROPS} />);
  expect(queryByTestId("input-title")).toBeDefined();
  expect(queryByTestId("input-description")).toBeDefined();
  expect(queryByTestId("action-delete")).toBeDefined();
  expect(queryByTestId("char-counter")).toBeDefined();
  expect(queryByTestId("update-success")).toBeDefined();
});

/**
 * ToDo: Update tests to use Enzyme
 * The tests below fail because react-testing-library is unable to focus the body, which is a necessary condition for
 * displaying the counter. https://github.com/kentcdodds/react-testing-library/issues/276#issuecomment-460962901
 **/

// it("displays the trash icon on hover", () => {
//   const { queryByTestId, container } = render(<Idea {...IDEA_PROPS} />);
//   expect(queryByTestId("action-delete")).toHaveStyle("opacity: 0");
//   fireEvent.mouseOver(container);
//   expect(queryByTestId("action-delete")).toHaveStyle("opacity: 1");
// });

// it("displays the character counter when body legnth exceeds threshold", () => {
//   const { queryByTestId, rerender } = render(<Idea {...IDEA_PROPS} />);
//   const updatedProps = {
//     ...IDEA_PROPS,
//     description:
//       "This is a description that will be used to test the total number of characters, and display a counter in the bottom left when it is close to max"
//   };
//
//   expect(queryByTestId("char-counter")).toHaveStyle("visibility: hidden");
//   rerender(<Idea {...updatedProps} />);
//   fireEvent.focus(queryByTestId("input-description"))
//   expect(queryByTestId("char-counter")).toHaveStyle("visibility: visible");
// });
