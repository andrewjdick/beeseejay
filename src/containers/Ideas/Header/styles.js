import styled from "styled-components";
import { PlusIcon } from "components/icons/PlusIcon";

export const HeaderWrapper = styled.header`
  display: flex;
  width: 100%;
  border-bottom: 1px solid rgb(239, 239, 239);
  background-color: white;
`;

export const HeaderInnerWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  max-width: 1440px;
  width: 100%;
  margin: 0 auto;
  padding: 1em;
`;

export const Logo = styled.img`
  width: 40px;
  height: auto;
`;

export const StyledPlusIcon = styled(PlusIcon)`
  &:hover {
    transform: rotate(90deg);
  }

  &:active {
    transform: rotate(90deg) scale(0.8);
  }
`;
