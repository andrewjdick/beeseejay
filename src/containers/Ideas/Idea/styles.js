import styled from "styled-components";
import { TextArea } from "components/inputs/TextArea";
import { TrashIcon } from "components/icons/TrashIcon";
import { media } from "media.js";

const CARD_DIMENSION = 150;

export const StyledTrashIcon = styled(TrashIcon)`
  opacity: 0;
  pointer-events: none;
  transition: opacity 100ms linear;
  padding: 10px;
  box-sizing: content-box;

  ${media.tablet`
    padding: 0;
  `};
`;

export const Container = styled.div`
  width: 100%;
  height: ${CARD_DIMENSION * 1.5}px;
  box-shadow: rgba(200, 200, 200, 0.5) 0px 2px 4px;
  border-radius: 5px;
  margin: 10px;
  padding: 10px;
  display: flex;
  flex-direction: column;
  background-color: whitesmoke;
  position: relative;

  &:hover {
    ${StyledTrashIcon} {
      opacity: 1;
      pointer-events: auto;
    }
  }

  &:last-child {
    margin-bottom: 2em;
  }

  ${media.tablet`
    width: ${CARD_DIMENSION}px;
    height: ${CARD_DIMENSION}px;

    &:last-child {
      margin-bottom: 0;
    }
  `}
`;

export const StyledTextArea = styled(TextArea)`
  flex: 1;
  min-width: 100%;
  max-width: 100%;
  resize: none;
  margin-top: 5px;
  padding: 5px;
`;

export const Counter = styled.span`
  padding: 10px;
  font-size: 1.1rem;
  color: ${({ charactersRemaining }) =>
    charactersRemaining <= 5 ? "red" : "orange"};

  ${media.tablet`
    padding: 0 5px;
  `};
`;

export const InfoWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  min-height: 25px;

  ${media.tablet`
    margin-top: 5px;
  `};
`;
