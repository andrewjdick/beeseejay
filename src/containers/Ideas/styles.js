import styled from "styled-components";
import { Select } from "components/inputs/Select";
import { media } from "media.js";

export const IdeaContainer = styled.section`
  flex: 1;
  display: flex;
  justify-content: center;
  overflow-y: scroll;
  overflow-x: hidden;

  /* Ensures momentum scrolling on iOS */
  -webkit-overflow-scrolling: touch;

  ${media.tablet`
    display: block;
  `}
`;

export const IdeaInnerContainer = styled.div`
  display: flex;
  flex-wrap: wrap;
  max-width: 1440px;
  margin: 0 auto;
  padding: 1em;
`;

export const UtilityContainer = styled.div`
  display: flex;
  justify-content: center;
  max-width: 1440px;
  width: 100%;
  margin: 0 auto;
  padding: 1em;

  ${media.tablet`
    justify-content: flex-end;
  `}
`;

export const SelectContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: baseline;

  ${media.tablet`
    flex-direction: row;
  `}
`;

export const SelectLabel = styled.label`
  margin-right: 10px;
  margin-bottom: 10px;
`;

export const StyledSelect = styled(Select)`
  position: relative;
  top: -2px;
`;
