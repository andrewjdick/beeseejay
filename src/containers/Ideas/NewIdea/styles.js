import styled from "styled-components";
import { PlusIcon } from "components/icons/PlusIcon";
import { baseCardStyles } from "containers/Ideas/Idea/styles";
import { media } from "media.js";

export const StyledPlusIcon = styled(PlusIcon)`
  color: rgba(200, 200, 200, 0.6);
`;

export const Container = styled.button`
  ${baseCardStyles};

  display: none;
  outline: none;
  border: 1px dashed rgba(200, 200, 200, 0.6);
  background-color: transparent;
  transition: all 200ms linear;

  &:hover {
    border-color: rgba(200, 200, 200, 0.8);
    box-shadow: 0px 0px 5px 2px rgba(200, 200, 200, 0.2);

    ${StyledPlusIcon} {
      color: rgba(200, 200, 200, 0.8);
    }
  }

  &:active {
    border-color: rgb(200, 200, 200);
    transform: scale(0.95);

    ${StyledPlusIcon} {
      color: rgb(200, 200, 200);
    }
  }

  ${media.tablet`
    display: inline-block;
  `};
`;
