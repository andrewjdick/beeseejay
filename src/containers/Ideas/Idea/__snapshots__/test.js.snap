// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<div
  className="sc-EHOje jQfXPD"
>
  <input
    className="sc-bwzfXH eMpGSe"
    readOnly={true}
    type="hidden"
    value="1234-abcd-5678-efgh-9101"
  />
  <input
    className="sc-bwzfXH eMpGSe"
    readOnly={true}
    type="hidden"
    value={1549985336574}
  />
  <input
    autoFocus={true}
    className="sc-bZQynM kihKbR sc-bwzfXH eMpGSe"
    data-testid="input-title"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="Enter a title..."
    type="text"
    value="Test title"
  />
  <textarea
    className="sc-gzVnrw eNaKsR sc-htpNat eoidjl"
    data-testid="input-description"
    maxLength="150"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Enter a description..."
    value=""
  />
  <div
    className="sc-dnqmqq gxdubg"
  >
    <button
      className="sc-ifAKCX jbICHU sc-bxivhb ekczNV"
      data-testid="action-delete"
      onClick={[Function]}
    >
      <svg
        className="sc-bdVaJa iKBuLo"
        fill="none"
        height={20}
        id="i-close"
        stroke="currentcolor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        viewBox="0 0 32 32"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M28 6 L6 6 8 30 24 30 26 6 4 6 M16 12 L16 24 M21 12 L20 24 M11 12 L12 24 M12 6 L13 2 19 2 20 6"
        />
      </svg>
    </button>
    <svg
      className="sc-iwsKbI jWofvB sc-bdVaJa iKBuLo"
      data-testid="update-success"
      fill="none"
      height={0}
      id="i-close"
      stroke="currentcolor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      viewBox="0 0 32 32"
      width={0}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M2 20 L12 28 30 4"
      />
    </svg>
    <span
      className="sc-htoDjs cdhgkb"
      data-testid="char-counter"
    >
      150
    </span>
  </div>
</div>
`;
